### ISSUE TITLE
'Authorize' or 'Authorization' not valid in JSON path

### ISSUE THREAD

**Issue by adamsproul** [gregsdennis/json-everything#211]
**Environment**
- Nuget Package: JsonPath.Net
- Nuget Version: 0.1.2
- OS: Windows 10
- .Net Target: .NET 6.0

**Describe the bug**
Any usage of the phrase 'Authorize' or 'Authorization' returns ```false``` when calling ```JsonPath.TryParse```

**To Reproduce**
In the immediate window or in a unit test the following give different results:

**Failed TryParse**
```var result = JsonPath.TryParse("$.authorize.sms.title", out var path)```
```result = false```

**Successful TryParse - changed 'z' to 's'**
```var result = JsonPath.TryParse("$.authorise.sms.title", out var path)```
```result = true```

**Expected behavior**
Both are valid JSON paths and ```TryParse``` should return true for any usage of the phrase 'authorize' or 'authorization'

**Comment by gregsdennis** [comment]
Wow that's a weird one. Thanks for reporting!  I'll set up a test and have a look.

**Comment by gregsdennis** [comment]
@adamsproul what are your local culture settings?  Both `authorize` and `authorise` work for me.

I'm not sure how this would fail.  I'm not doing any kind of spell check or known word check.  It's literally parsing the property name from between the `.`s.

**Comment by adamsproul** [comment]
@gregsdennis  - I could not replicate the behavior in its own project, I did however look to see if there was more detail by just doing a Parse where the original exception is occurring and I get the below exception.

`
JsonPath.Parse("$.authorize.sms.title")

'JsonPath.Parse("$.authorize.sms.title")' threw an exception of type 'Json.Path.PathParseException'
    Data: {System.Collections.ListDictionaryInternal}
    HResult: -2146233088
    HelpLink: null
    Index: 9
    InnerException: null
    Message: "Could not identify selector"
    Source: "JsonPath.Net"
    StackTrace: "   at Json.Path.JsonPath.Parse(String source)"
    TargetSite: {Json.Path.JsonPath Parse(System.String)}
`

That looks like it is the 'z' character it doesn't like. I tried using a virtual keyboard to enter the z in case there was anything weird with it and I still see the same behavior. Wherever the 'z' character goes the exception index changes.

It does not seem to be a limitation with the word 'authorize' but instead something strange in our code that is treating 'z' differently.

We can probably close this ticket for now, and if I can find a good way to replicate it I will re-open.

I appreciate your help
